### PR TITLE
[go_expvar] use AgentCheckTest & add CI

### PR DESCRIPTION
**Pull request by degemer**
- clean mocked tests (use AgentCheckTest now)
- add a ci with a real go program (Travis boxes contains a go install)
- a-z in Rakefile import
- usual pep8 cleaning

### CODE REVIEW

**Comment by degemer** [comment]
Both `go_expvar` tests passed.

File: ci/resources/go_expvar/test_expvar.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+lastUser.Set(user)
+
+msg := fmt.Sprintf("G'day %s\n", user)
+	io.WriteString(w, msg)
+}
+
+func main() {
+	http.HandleFunc("/", HelloServer)
+	http.ListenAndServe(":8079", nil)
+}
```
Comment (LeoCavaille) [inline ci/resources/go_expvar/test_expvar.go]: very very nitpicky, but this could `go fmt`-d

Comment (degemer) [inline ci/resources/go_expvar/test_expvar.go]: Sure, updating (bad c/p)

File: tests/test_go_expvar.py [hunk 2]
```diff
@@ -1,152 +1,224 @@
...
-    def setUp(self):
-        self.agentConfig = {
-            'version': '0.1',
-            'api_key': 'toto'
+def _get_data_mock(url):
+    with open(url, 'r') as go_output:
+        return json.loads(go_output.read())
+
+
+@attr(requires='go_expvar')
```
Comment (LeoCavaille) [inline tests/test_go_expvar.py]: If only one of the test cases requires the integration setup, you should put the `attr` on those test cases. This allows people to run unit tests quickly by running `ci:run`.

File: tests/test_go_expvar.py [hunk 3]
```diff
@@ -1,152 +1,224 @@
-import unittest
+from collections import defaultdict
+import copy
+import os.path
 import time
-import os
-from tests.common import load_check
-import simplejson as json
 
+from nose.plugins.attrib import attr
```
Comment (remh) [inline tests/test_go_expvar.py]: nitpick but you should arrange your imports here :)

**Comment by LeoCavaille** [comment]
Merging, was originally failing on supervisor tests.

**Pull request merged**